### DISCUSSION TITLE
tutorials/performance/vertex_animation/animating_thousands_of_fish

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#9] [User-contributed notes]
# tutorials/performance/vertex_animation/animating_thousands_of_fish

This tutorial explores a technique used in the game ABZU for rendering and animating thousands of fish using vertex animation and static mesh instancing. In Godot, this can be accomplished with a c...

https://docs.godotengine.org/en/latest/tutorials/performance/vertex_animation/animating_thousands_of_fish.html

**Comment by baz-atak** [comment]
To animate the school of fish using INSTANCE_CUSTOM values, you need to enable their use on the MultiMesh by setting use_custom_data to true. This needs to be done before setting the instance_count.

**Comment by tobique-gd** [comment]
You can move the fishes with a compute shader by passing the fish's multimesh buffer RID to the compute shader and operating on it. That way you can take advantage of the GPU while having infinite lifetime and also more control in the shader.
